src/math_tools/gyrokinetic_poisson_bracket.hpp

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include "ddc_alias_inline_functions.hpp"
+#include "ddc_aliases.hpp"
+#include "gradient.hpp"
+#include "metric_tensor_evaluator.hpp"
+#include "static_tensors.hpp"
+#include "vector_field.hpp"
+
+
+/**
+ * @brief A class which implements a gyrokinetic Poisson bracket operator.
+ * The implemented equation is:
+ * @f$ \{F, G\} = b\dot(\nabla F \cross \nabla G) @f$
+ * with @f$ b= \mathbf{B} / B @f$ the unitary magnetic field, i.e:
+ * @f$ \{F, G\} = {\cal J}_{\rm x}^{-1}\epsilon^{ijk}\partial_{x^i} F \partial_{x^j} G b_k @f$
+ * with @f$ {\cal J}_{\rm x} @f$ the jacobian of the system,
+ * @f$ b_k @f$ the covariant components of b and @f$\epsilon^{ijk} @f$ the Levi-Civita symbol.
+ * @tparam Mapping3D A type representing a mapping in 3 dimensions.
+ */
+template <class Mapping3D>
+class GyrokineticPoissonBracket
```

Comment:
I am not convinced that this is the right name for this class. See the [discussion](https://github.com/gyselax/gyselalibxx/issues/15#issuecomment-2845087397) on #15 .

Maybe it should be called:
```suggestion
class LiePoissonBracket
```
?